### ISSUE TITLE
keras.callbacks.tensorboard compute gradient histogram using validation data

### ISSUE THREAD

**Issue by LihaoWang1991** [keras-team/keras#13204]
When I am using the tensorboard function in keras.callbacks, I find that the output gradient histogram is computed using validation dataset, which is not normal (it should be computed using training dataset). Here is how I find it:

**The definition of the tensorboard function in my case:**

tensorboard_callback = keras.callbacks.TensorBoard(log_dir = './experience_2', histogram_freq=1, write_grads=True, batch_size=512, write_images= True, update_freq='batch')

model.fit(X_train, y_train, batch_size=512, nb_epoch=1, verbose=1, validation_data=(X_val, y_val), callbacks=[self.tensorboard_callback])

**Test cases:** 

**case 1:**
size of X_train is [512, 5],    (5 is the input dimension in my model)
size of y_train is [512, 1]      (1 is the input dimension in my model)

size of X_val is [512, 5],    
size of y_val is [512, 1]      

The gradient of each layer in the ouput histogram has only 1 curve (this is normal)

**case 2:**
size of X_train is [512, 5],   
size of y_train is [512, 1]      

size of X_val is [1024, 5],    
size of y_val is [1024, 1]      

The gradient of each layer in the ouput histogram has 2 curves

**case 3:**
size of X_train is [1024, 5],   
size of y_train is [1024, 1]      

size of X_val is [512, 5],    
size of y_val is [512, 1]      

The gradient of each layer in the ouput histogram has only 1 curve

So I believe that the gradient histogram is computed using validation data, which is not normal. Because the gradient is computed only during training.
